lib/util/source-code.js

```diff
@@ -163,6 +163,16 @@ SourceCode.prototype = {
     },
 
     /**
+     * Retrieves an array containing all tokens and comments in the source code sorted by range.
+     * @returns {ASTNode[]} An array of tokens and comments sorted by range start.
+     */
+    getAllTokensAndComments: function() {
```

Comment:
It's better to do this once and store on SourceCode. It's never going to change.